### ISSUE TITLE
Unhandled error on native side

### ISSUE THREAD

**Issue by danielrhodes** [dominicstop/react-native-ios-context-menu#94]
I am seeing these come up as unhandled errors in my exception tracker in production. I am on v 2.3 of this library and v 4.2.3 of react-native-ios-utilities. It seems to happen regardless of iOS version or device or React Native version (recently upgraded to 0.73 from 0.72).

I suspect what is happening is some sort of re-render is happening and the context menu loses its underlying reference to the root view.

```
Error domain: react-native-ios-utilities - code: viewNotFoundForReactTag - description: No corresponding view instance found for react tag - extraDebugValues: extraDebugValuesString: { node: 597, targetType: RNIContextMenuView } - fileName: /Users/<REDACTED>/code/<REDACTED>/packages/app/node_modules/react-native-ios-utilities/ios/Sources/Helpers/RNIModuleHelpers.swift - functionName: getView(withErrorType:forNode:type:bridge:) - lineNumber: 39 - columnNumber: 27 
    (native) construct
    (native) apply
    /Users/<REDACTED>/Library/Developer/Xcode/DerivedData/app-elacbqvdwmqsunameeuzcynrbwae/Build/Intermediates.noindex/ArchiveIntermediates/app/BuildProductsPath/Release-iphoneos/main.jsbundle:3211:28 _construct
    /Users/<REDACTED>/Library/Developer/Xcode/DerivedData/app-elacbqvdwmqsunameeuzcynrbwae/Build/Intermediates.noindex/ArchiveIntermediates/app/BuildProductsPath/Release-iphoneos/main.jsbundle:3173:25 Wrapper
    (native) construct
    /Users/<REDACTED>/Library/Developer/Xcode/DerivedData/app-elacbqvdwmqsunameeuzcynrbwae/Build/Intermediates.noindex/ArchiveIntermediates/app/BuildProductsPath/Release-iphoneos/main.jsbundle:187182:322 _createSuperInternal
    (native) call
    /Users/<REDACTED>/Library/Developer/Xcode/DerivedData/app-elacbqvdwmqsunameeuzcynrbwae/Build/Intermediates.noindex/ArchiveIntermediates/app/BuildProductsPath/Release-iphoneos/main.jsbundle:187195:26 CodedError
```

which references https://github.com/dominicstop/react-native-ios-utilities/blob/master/ios/Sources/Helpers/RNIModuleHelpers.swift#L38-L50

Is there anything I can do to mitigate this?

**Comment by valentinchelle** [comment]
I've noticed this issue too after upgrading to 0.73. 
Additionally, I noticed that the Auxiliary preview crashes when dismissed.

**Comment by danielrhodes** [comment]
@valentinchelle I haven't been able to consistently reproduce it locally, but previously I found that it was quite vulnerable to remounts of the parent component / tree and fiddling with the cleanup may have helped somewhat.

**Comment by nandorojo** [comment]
Confirming we are getting the same issue:

![image](https://github.com/dominicstop/react-native-ios-context-menu/assets/13172299/3a205ec8-b567-47b7-8e6d-badb24f4a50a)

**Comment by dominicstop** [comment]
i'll take a look and start debugging

**Comment by alantoa** [comment]
I encountered this issue before, and it was fixed by `react-native-ios-context-menu` v2.4.0 and `react-native-ios-utilities` v4.3.0 on my end. 
Could you guys try it out?

**Comment by danielrhodes** [comment]
This is still occurring for me, but I might have a better clue about what is happening:

I am using react-navigation, and if I click a menu item which triggers a screen change, it might cause this error. I believe this is because the menu has not closed by the time the view is removed.

**Comment by dominicstop** [comment]
@danielrhodes hi, i need some sort of small repro/code snippet so i can debug this further (i'll add it as a test case in the example app for the future). i'm having some trouble trying to recreate the issues you're describing.

if possible, can you please provide some screenshots/video of the performance monitor while re-creating the issue/bug? (you can crop the video via the photos app to crop/redact the contents of your app)

**Comment by danielrhodes** [comment]
@dominicstop I haven't been able to replicate it locally, but I am seeing it quite often from exception logs in production.

Here is an example project that isn't much different from my own to give you something to work with. https://github.com/danielrhodes/ios-context-menu-bug-example

**Comment by alantoa** [comment]
yeah, we only found this error log in the production log. However, so far I have discovered that this error has not caused the app to crash yet. so it's kind of weird...

**Comment by dominicstop** [comment]
@danielrhodes hello, i've done some more research...

i look at the [code](https://github.com/facebook/react-native/blob/12f2c3cd2bbf186afe1b4052ebf377d8030c7a73/packages/react-native/React/CoreModules/RCTPerfMonitor.mm#L417-L425) for `RCTPerformanceManager`, and it looks like it just counts the views from the view registry....

<img width="1552" alt="2024-02-26-11-21-15" src="https://github.com/dominicstop/react-native-ios-context-menu/assets/18517029/15e682a3-7d06-40f9-bbbb-3a0d8539a7e6">

after that, i started debugging again...

* removed all the views in the JS-side via conditional rendering
* fired up the memory graph; it's saying there's 11 `RCTView` instances in memory
* read `RCTUIManager._viewRegistry`, it's saying there's 1584 items...
* But all the ref. were held by `REAUIManager` from `react-native-reanimated`?
* `REAUIManager` is a custom subclass of `RCTUIManager`, but i don't really know how it's connected? 
<img width="1373" alt="2024-02-26-11-07-40" src="https://github.com/dominicstop/react-native-ios-context-menu/assets/18517029/fd277130-7701-4e33-b013-d830d106ccec">

<br><br>

i don't really understand why `REAUIManager._viewRegistry` is holding all those views?

**Comment by nandorojo** [comment]
Interestingly, I don't think I've had this issue since upgrading to 2.4.2 and ios-utilities 4.3.1

**Comment by danielrhodes** [comment]
> EDIT: I think i might finally [fixed this](https://github.com/dominicstop/react-native-ios-context-menu/issues/94#log-2024-02-27-19-05) in the latest [prerelease](https://github.com/dominicstop/react-native-ios-context-menu/compare/v2.5.0-1...v2.5.0-2) version.

Nice! When the release comes out, I can cut new release to try it.

**Comment by dominicstop** [comment]
hello @danielrhodes  

for now you would have to use: `react-native-ios-context-menu@v2.5.0-2` + `react-native-ios-utilities@v4.4.0-5`.

i'm still testing all of the example + test items in the example; i just wanna be sure first that the changes i've made recently won't cause a crash somehow (i haven't found any issues so far...)

it would be great if you could try it out, and see if there are any bugs + give feedback ahsdjhkdsldfkdj

if you decide to install the pre-release version, i recommend adding the code snippet below in you `App.js`/app entry file so we can get some more debugging info (you can disable/remove them anytime).

```js
import { setSharedEnvForRNICleanableViewRegistry, setSharedEnvForRNIUtilitiesModule } from 'react-native-ios-utilities';

setSharedEnvForRNIUtilitiesModule({
  debugShouldLogViewRegistryEntryRemoval: true,
  overrideEnableLogStackTrace: true,
  overrideShouldLogFileMetadata: true,
  overrideShouldLogFilePath: true
});

setSharedEnvForRNICleanableViewRegistry({
  debugShouldLogCleanup: true,
  debugShouldLogRegister: true,
  shouldGloballyDisableCleanup: false,
});
```
